### PR TITLE
Update Apache Airflow version

### PR DESCRIPTION
**Pull request by obulat**
## Description

This PR updates Apache Airflow version from 2.0.2 to 2.1.2.

## Technical details

Together with Apache Airflow, other dependency packages are also updated. This PR also introduces using [a constraints file](https://pip.pypa.io/en/stable/user_guide/#constraints-files) from Apache Airflow repository which sets the specific versions of packages that should be installed to ensure correct build. With this, we can clean up the `requirements.txt` files to only include top-level packages and versions of the packages that are not used by Airflow.

I have also considered replacing `psycopg2-binary` with `psycopg2` as it [was done in the API](https://github.com/WordPress/openverse-api/pull/183). However, since Airflow installs `psycopg2-binary` anyways, it is probably unnecessary.

## Tests

To test this PR, you can rebuild and run the webserver container and make sure that DAGs can be loaded and run in the Web UI. You can also run the tests in the local Docker container.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main` or `master`).
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: openverse_catalog/Dockerfile [hunk 1]
```diff
@@ -4,6 +4,8 @@ ENV AIRFLOW_HOME=/usr/local/airflow
 ARG OUTPUT_DIR=/tmp/workflow_output/
 ENV PATH=/usr/local/airflow/.local/bin:$PATH
 ARG REQUIREMENTS_FILE=requirements_prod.txt
+ARG CONSTRAINTS_FILE="https://raw.githubusercontent.com/apache/airflow/constraints-2.1.2/constraints-3.9.txt"
```
Comment (zackkrida) [inline openverse_catalog/Dockerfile:7]: Great!

**Review by zackkrida — APPROVED** [review]
This is so cool, such an improvement. Builds and tests pass locally.

File: openverse_catalog/Dockerfile (hunk 1 above)
Comment (krysal) [inline openverse_catalog/Dockerfile:7]: Referencing the documents can also be helpful.
```suggestion
ARG REQUIREMENTS_FILE=requirements_prod.txt
# https://airflow.apache.org/docs/apache-airflow/2.1.2/installation.html#constraints-files
ARG CONSTRAINTS_FILE="https://raw.githubusercontent.com/apache/airflow/constraints-2.1.2/constraints-3.9.txt"
```

**Review by krysal — APPROVED** [review]
Main dependencies are clearer now, of course. Thanks!

File: openverse_catalog/Dockerfile (hunk 1 above)
Comment (obulat) [reply openverse_catalog/Dockerfile:7]: Definitely, @krysal ! Thank you :)

**Pull request merged**